### PR TITLE
Update proposal action endpoint's CREATE_CIRCLE handler

### PR DESCRIPTION
**Pull request by danielr18**
https://github.com/daostack/common-governance-office/issues/2
https://github.com/daostack/common-governance-office/issues/3

### CODE REVIEW

File: functions/src/proposals/controllers/proposals/action.ts [hunk 1]
```diff
@@ -27,11 +28,16 @@ export const action = async (req: Request, res: Response, next: NextFunction) =>
...
+            let isAllowed = false;
+            if (req.user?.roles?.includes(USER_ROLES.ADMIN)) {
+                isAllowed = true;
+            } else {
+                const commonMember = await CommonMembersSubcollectionDb.get({ userId: req.user.uid, commonId: args?.commonId }, true)
+                const governance = await governanceDb.get({ commonId: args.commonId }, true)
+                isAllowed = Boolean(hasPermission({ commonMember, circles: governance.circles, key: type }))
+            }
+          
+            if (!isAllowed) throw new CommonError("User not allowed", { type, commonId: args.commonId })
```
Comment (danielr18) [inline functions/src/proposals/controllers/proposals/action.ts:40]: This is allowing an admin to execute these actions regardless of their membership:

* CREATE_CIRCLE
* LEAVE_CIRCLE
* CREATE_PROJECT
* INVITE_TO_CIRCLE
* INVITE_TO_COMMON
* PIN_OR_UNPIN_FEED_ITEMS

@elatif2020 Is this ok? should it only allow CREATE_CIRCLE?

**Review by ariel-dao — APPROVED** [review]

File: functions/src/proposals/controllers/proposals/action.ts (hunk 1 above)
Comment (elatif2020) [reply functions/src/proposals/controllers/proposals/action.ts:40]: yes
and we might want to allow more actions in the future such as
UPDATE_COMMON
and also (maybe more complicated)
DELETE_COMMON

**Comment by ariel-dao** [comment]
@danielr18 can we merge this? just update with latest dev branch please

**Comment by danielr18** [comment]
@ariel-dao done

**Comment by ariel-dao** [comment]
Congratulations @danielr18  :) First PR is in!

**Pull request merged**